### DISCUSSION TITLE
Sidebar not updated upon document publication

### DISCUSSION THREAD

**Discussion by westong8** [outline/outline#2965] [Self Hosting]
**To Reproduce**
1. Using latest Outline docker image in self-hosted production.
2. Also utilizing Redis and Postgres servers.
3. `yarn start --services=web,websockets` & `yarn start --services=worker` after images are loaded built.

**Expected behavior**
Collection information should be updated upon the publication of a document. Would assume it's a websocket error, but no errors related to websockets are logged. New documents are only displayed on sidebar after refreshing the page and everything else runs as expected. Any suspicions on where this issue could be coming from?

**Outline (please complete the following information):**
- Install: self hosted
- Version: docker image: sha256:2048ed0e0b7c0ed22210f124d9d8df628864066f8104f0b742b60c044629ef6b     

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: chrome
 - Version 96.0.4664.110

**Comment by tommoor** [comment]
Could you post a screenshot showing the active websocket connection in the Chrome inspector? Many thanks.

**Reply by westong8** [reply]
Hey, thanks for the quick correspondence. I've attached two images below, 1 showing the active websocket connection and another displaying the messages sent and received during the creation and publication of the doc. Please let me know if any other pieces of info are needed.
<img width="1253" alt="Screen Shot 2022-01-21 at 9 26 40 AM" src="https://user-images.githubusercontent.com/93536162/150572501-dcaa722a-8466-4ad9-9d95-ea08b7d217b0.png">
<img width="1230" alt="Screen Shot 2022-01-21 at 9 25 40 AM" src="https://user-images.githubusercontent.com/93536162/150572510-cdb901ec-73f2-4818-b67e-4bea154d61a3.png">
.

**Reply by tommoor** [reply]
Good to see confirmation that it's connected and that collaboration is not enabled. Perhaps you can monitor the websocket messages while publishing a doc to see if that offers any clues?

**Reply by westong8** [reply]
I have attached images showing the messages for the expected behavior in which the newly-created document is immediately reflected in the side bar versus the messages sent in the self-hosted version (issue in top image, expected behavior on bottom image). It seems that the self-hosted version doesn't send the request for `documents.publish`. Does this suggest any potential source for the issue?

<img width="1249" alt="Screen Shot 2022-01-21 at 11 35 57 PM" src="https://user-images.githubusercontent.com/93536162/150629582-d81b34f4-ac3e-4c4f-971d-ab80d2d59200.png">
<img width="1249" alt="Screen Shot 2022-01-21 at 11 38 36 PM" src="https://user-images.githubusercontent.com/93536162/150629584-f630d82a-9216-4518-a6b1-dae107e5e8d4.png">

**Reply by westong8** [reply]
@tommoor Seems that none of the "entities" events are emitted despite the websocket connection being present for all actions such as document creation or publication.

**Reply by tommoor** [reply]
That would cause the problem, it most likely would mean that your socket is not subscribed to the collection channel but I don't know how that could happen where you could see the collection in app but not be subscribed on the backend.

One way to check if this is the case would be to open two tabs, edit the name of the collection in one and see if it updates in the other.

**Reply by westong8** [reply]
Just edited the name of the collection in one tab and the updates were not reflected on the other. Update was only reflected after reloading the page.

**Reply by westong8** [reply]
I'd also like to add that I'm using Heroku to both deploy the instance and handle the Postgres and Redis servers. Not sure if that may also affect it.

**Reply by tommoor** [reply]
As long as it's connected you're good in regards to hosting environment. I would like to get to the bottom of this but cannot reproduce in any other environment – how do the collection permissions look?

**Reply by westong8** [reply]
Collection permissions work as expected (changing permissions, adding groups and members, editing member permissions). Is there something I should be looking out for?

Additionally, not sure if this may reveal anything, but the document publication seems to be the only thing not working normally as any edits to the title of a document are reflected immediately despite the "entities" message not going through.

Looked through the code but are there any specific conditions for the "entities" emissions in particular?

**Reply by tommoor** [reply]
Sorry, I mean what are the permissions on the collection.

> are there any specific conditions for the "entities" emissions in particular?

No, I believe you're not being subscribed to the websocket channel for the collection

**Reply by westong8** [reply]
Currently "No access" with public document sharing disabled. Issue persists for "View only" and "View and Edit" access.

**Reply by tommoor** [reply]
Could you try running the worker in the same process?
